explorer/websockethandlers.go

```diff
@@ -106,20 +106,29 @@ func (exp *explorerUI) RootWebsocket(w http.ResponseWriter, r *http.Request) {
 					webData.Message = string(msg)
 
 				case "getticketpooldata":
-					// TODO: Implement a cache to control websocket connections
-					// meant to serve the same data.
-					cData, gData, err := exp.explorerSource.TicketPoolVisualization(
-						dbtypes.ChartGroupingFromStr(msg.Message),
-					)
-					if err != nil {
-						if strings.HasPrefix(err.Error(), "unknown interval") {
-							log.Debugf("Invalid ticket pool interval provided via getticketpooldata: %s", msg.Message)
-							webData.Message = "Error: " + err.Error()
+					var err error
+					interval := dbtypes.ChartGroupingFromStr(msg.Message)
+
+					cData, gData, ok := GetTicketPoolData(interval)
```

Comment:
There's still a gotcha with this approach.  What happens when there's another handler already running `TicketPoolVisualization`, but the cache is still out-of-date when a second client requests ticket pool data?  Any subsequent handlers will also query the DB.  This is what all the trylock and ok to update stuff is for in the DevFundBalance type.

Reply:
So if I get your point trylock in this case scenario is supposed to block consecutive calls made to `TicketPoolVisualization` method till its first instance is resolved?